test/Microsoft.IdentityModel.JsonWebTokens.Tests/JwtTokenUtilitiesTests.cs

```diff
@@ -208,6 +215,281 @@ public void ResolveTokenSigningKey()
             Assert.Null(resolvedKey);
         }
 
+        #region DecryptJwtToken Tests
+
+        [Fact]
+        public void GetCryptoProviderFactory_WhenSetOnTokenValidationParameters_ReturnsFromValidationParameters()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters()
+            {
+                CryptoProviderFactory = new CryptoProviderFactory(),
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Same(validationParameters.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void GetCryptoProviderFactory_WhenSetToNullOnTokenValidationParameters_ReturnsFromSecurityKey()
+        {
+            // Arrange
+            // SecurityKey can never have 'null' CryptoProviderFactory.
+            var securityKey = symmetricSecurityKey;
+            var validationParameters = new TokenValidationParameters
+            {
+                CryptoProviderFactory = null,
+            };
+
+            // Act
+            var result = JwtTokenUtilities.s_getCryptoProviderFactory(validationParameters, securityKey);
+
+            // Assert
+            Assert.NotNull(result);
+            Assert.Equal(securityKey.CryptoProviderFactory, result);
+        }
+
+        [Fact]
+        public void DecryptJwtToken_WhenValidationParametersIsNull_ShouldThrow()
+        {
+            // Arrange
+            TokenValidationParameters validationParameters = null;
+
+            // Act & Assert
+            var exception = Assert.Throws<ArgumentNullException>(() =>
+                JwtTokenUtilities.DecryptJwtToken(
+                    securityToken: null,
+                    validationParameters: validationParameters,
+                    decryptionParameters: null));
+
+            Assert.Equal("validationParameters", exception.ParamName);
```

Comment:
Use `nameof()`.

Reply:
I purposely have used literal string in the ``Assert`` on lines 271, and 288 (pasted below). 

``
Assert.Equal("validationParameters", exception.ParamName);
``
``
Assert.Equal("decryptionParameters", exception.ParamName);
``

I didn't use ``nameof`` as the ``exception.ParamName`` is set to the corresponding parameter name of the ``DecryptJwtToken`` function. If those parameters' names change, this test will break and require updating the literal string.

If I use the suggested nameof(), I will set have to update the variable name in the test to match the parameter name of the ``DecryptJwtToken`` function, though not the nameof().

That said, I moved the strings to a variable. That, I think is cleaner and also does not use literal inline in ``Assert``.